libs/sdk-ui-dashboard/src/model/commandHandlers/drillTargets/test/addDrillTargetsHandler.test.ts

```diff
@@ -0,0 +1,37 @@
+// (C) 2021 GoodData Corporation
+import { uriRef } from "@gooddata/sdk-model";
+import { DashboardTester, preloadedTesterFactory } from "../../../tests/DashboardTester";
+import { addDrillTargets } from "../../../commands/drillTargets";
+import { DrillTargetsAdded } from "../../../events/drillTargets";
+import { SimpleDashboardIdentifier } from "../../../tests/Dashboard.fixtures";
+import { IAvailableDrillTargets } from "@gooddata/sdk-ui";
+import { selectDrillTargetsByWidgetRef } from "../../../state/drillTargets/drillTargetsSelectors";
+
+describe("addAttributeFilterHandler", () => {
+    const widgetRef = uriRef("someuri");
+    const availableDrillTargetsMock: IAvailableDrillTargets = {};
+
+    let Tester: DashboardTester;
+    beforeEach(
+        preloadedTesterFactory((tester: DashboardTester) => (Tester = tester), SimpleDashboardIdentifier),
+    );
+
+    it("should add  drill target to the state for given widget", async () => {
+        const event: DrillTargetsAdded = await Tester.dispatchAndWaitFor(
+            addDrillTargets(widgetRef, availableDrillTargetsMock),
+            "GDC.DASH/EVT.DRILL_TARGETS.ADDED",
+        );
+        expect(event.payload.widgetRef).toEqual(widgetRef);
+        expect(event.payload.availableDrillTargets).toMatchSnapshot();
+
+        const drillTargets = selectDrillTargetsByWidgetRef(widgetRef)(Tester.state());
+        expect(drillTargets?.availableDrillTargets).toEqual(availableDrillTargetsMock);
+    });
+
+    it("should emit the appropriate events for add drill target", async () => {
+        Tester.dispatch(addDrillTargets(widgetRef, availableDrillTargetsMock));
```

Comment:
please dispatch command that also uses the `TestCorrelation` (from fixtures).. in combination with the emitted event digest, this will test that correlation is correctly propagated from command to the events.

Reply:
done